src/pages/index.js

```diff
@@ -66,40 +128,82 @@ profileEditButton.addEventListener("click", () => {
   editPopupForm.open();
 });
 
-// Function to render a card
+avatarEditButton.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
+//to render a card
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  section.addItem(card.getCardElement());
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    function handleDeleteClick() {
+      cardDeletePopup.setSubmitAction(() => {
+        cardDeletePopup.setLoading(true);
+        api
+          .removeCard(cardData._id)
+          .then((res) => {
+            console.log("here");
+            cardElement.removeCardElement(res._id);
+            cardDeletePopup.close();
+          })
+          .catch(console.error)
+          .finally(() => {
+            cardDeletePopup.setLoading(false);
+          });
+      });
+      cardDeletePopup.setEventListeners();
+      cardDeletePopup.open();
+    },
+    function handleLikeButtonClick() {
+      api
+        .changeLikeStatus(cardData._id, cardElement.isLiked)
+        .then((res) => cardElement.updateLikes(res.isLiked))
+        .catch(console.error);
+    }
+  );
+  // section.addItem(card.getCardElement());
+  return cardElement.getCardElement();
 }
 
-// Section for managing cards
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  cardsWrap
-);
-
-// Render initial cards
-section.renderItems();
-
 /* -------------------------------------------------------------------------- */
 /*                                  Handlers                                  */
 /* -------------------------------------------------------------------------- */
 
 // Event handler for profile edit form submission
 function handleEditProfileSubmit(formData) {
   const { nameInfo, jobInfo } = formData;
-  userInfo.setUserInfo(nameInfo, jobInfo);
-  editPopupForm.close();
+  editPopupForm.setLoading(true);
+  api
+    .setUserInfo(nameInfo, jobInfo)
+    .then(() => {
+      console.log("here");
+      userInfo.setUserInfo(nameInfo, jobInfo);
+      editPopupForm.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      editPopupForm.setLoading(false);
+    });
 }
 
 // Event handler for add card form submission
 function handleAddCardFormSubmit(inputValues) {
   const { name, link } = inputValues;
+  addCardPopupForm.setLoading(true);
+  api
+    .addCard({ name, link })
+    .then((cardData) => {
+      const cardElement = renderCard(cardData);
+      cardSection.prependItem(cardElement);
+      addCardPopupForm.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addCardPopupForm.setLoading(false);
+    });
   renderCard({ name, link });
```

Comment:
`renderCard({ name, link }); is not needed here because you create the card in line 198 in block `then`